packages/pagination/src/index.ts

```diff
@@ -176,6 +176,7 @@ export default defineComponent({
 
     watch(() => props.currentPage, val => {
       internalCurrentPage.value = getValidCurrentPage(val)
+      lastEmittedPage.value = internalCurrentPage.value
```

Comment:
This patch is wrong, which will introduce breaking new bug.
As your code, `lastEmittedPage.value` will equals `internalCurrentPage.value`, thus [`emitChange `](https://github.com/element-plus/element-plus/blob/dev/packages/pagination/src/index.ts#L122) will never be triggered(unless size changed).
Maybe L178 & L179 should have order exchanged and run test again?

Reply:
No, it's right.
`props.currentPage change` only happy in parent change it. The pagination only change `internalCurrentPage.value`, in L125 we can find ` lastEmittedPage.value = internalCurrentPage.value`, so if parent change currentPage, `lastEmittedPage.value` need to equals `currentPage.value`, and `emitChange` can work normal.

If don't add L179, it's work not normal. Details in https://github.com/element-plus/element-plus/issues/2025, it have a case to show the bug.

Reply:
Sorry, I checked again, I think you are right. I'll delete my previous comment so as not to confuse others.

Reply:
nothing, thanks for your check